app/models/generic_object_definition.rb

```diff
@@ -36,8 +36,8 @@ def properties=(props)
     super
   end
 
-  def type_cast(attr_name, value)
-    TYPE_MAP.fetch(defined_property_attributes[attr_name]).cast(value)
+  def valid_data_type?(attr_name, value)
+    value.kind_of?(TYPE_MAP.fetch(defined_property_attributes[attr_name]))
```

Comment:
Not sure I like this.  One nice thing of having the ActiveRecord types is the convertability.  That is, you can give a string to the integer type and it would convert correctly.  I still think type casting would be really nice to keep.